File: src/main/webapp/storage/src/app/dto/MovingRecordResponse.ts
import {Item} from "./Item";

export class MovingRecordResponse {
  private readonly _count: number;
  private readonly _date: string;
  private readonly _type: string;
  private readonly _username: string;
  private readonly _item: Item;

  constructor(count: number, date: string, type: string, username: string, item: Item) {
    this._count = count;
    this._date = date;
    this._type = type;
    this._username = username;
    this._item = item;
  }


  get count(): number {
    return this._count;
  }

  get date(): string {
    return this._date;
  }

  get type(): string {
    return this._type;
  }

  get item(): Item {
    return this._item;
  }

  get username(): string {
    return this._username;
  }
}

File: src/main/java/com/sonet/storage/dto/response/MovingRecordResponse.java
package com.sonet.storage.dto.response;

import com.sonet.storage.model.item.Item;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;

@AllArgsConstructor
@Getter
@Setter
public class MovingRecordResponse {

    private Long count;

    private String date;

    private String type;

    private String username;

    private Item item;
}
